manifests/key.pp

```diff
@@ -15,7 +15,7 @@
     command     => "/usr/sbin/dnssec-keygen -a HMAC-MD5 -r /dev/urandom -b 128 -n USER ${name}",
     cwd         => "${cfg_dir}/bind.keys.d",
     require     => [
-      Package['dnssec-tools','bind9'],
+      Package['dnssec-tools',$package],
```

Comment:
Yea, the dnssec-tools is the actual requirement, as attached to the exec. It doesn't actually need to depend on bind.